boa/src/syntax/lexer/error.rs

```diff
@@ -20,13 +22,26 @@ impl Error {
     {
         Self::Syntax(err.into())
     }
+
+    /// Creates a new StrictMode error.
+    ///
+    /// This error is used to represent the case where a piece of javascript
+    /// cannot be lexed/parsed because it is in invalid when strict mdoe is
+    /// enabled.
+    pub(super) fn strict<M>(err: M) -> Self
+    where
+        M: Into<Box<str>>,
+    {
+        Self::StrictMode(err.into())
+    }
```

Comment:
Could we add a link to where this error is defined in the spec?

Reply:
You mean https://tc39.es/ecma262/#sec-strict-mode-code + https://tc39.es/ecma262/#prod-annexB-LegacyOctalIntegerLiteral ?

Reply:
Checking https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Deprecated_octal

It seems it's just a normal syntax error, right?

Reply:
Doesn't this mean that we then can't differentiate what caused the syntax error? Would it perhaps be better to have the syntax error to have a caused by field or something?

Reply:
Removed as per spec